From 7b5182a572c03279b65f0c6abd434edd513e956d Mon Sep 17 00:00:00 2001
From: Thomas Blume <Thomas.Blume@suse.com>
Date: Fri, 7 Aug 2020 08:39:26 +0200
Subject: [PATCH] net-lib.sh: support infiniband network mac addresses

Infiniband MAC addresses have 20 octets.

Reference: bsc#996146
(cherry picked from commit 376ce85105121936666349aa5a777768d52516f7)

Resolves: #1964538
---
 modules.d/40network/net-lib.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index f0c1c041..128d6343 100755
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -570,6 +570,11 @@ parse_ifname_opts() {
             # udev requires MAC addresses to be lower case
             ifname_mac=$(echo $2:$3:$4:$5:$6:$7 | sed 'y/ABCDEF/abcdef/')
             ;;
+        21)
+            ifname_if=$1
+            # udev requires MAC addresses to be lower case
+            ifname_mac=$(echo $2:$3:$4:$5:$6:$7:$8:$9:${10}:${11}:${12}:${13}:${14}:${15}:${16}:${17}:${18}:${19}:${20}:${21} | sed 'y/ABCDEF/abcdef/')
+            ;;
         *)
             die "Invalid arguments for ifname="
             ;;

